LibraryManagementSystem/Forms/ViewTab.cs

```diff
@@ -70,13 +89,27 @@ private void btnSearch_Click(object sender, EventArgs e)
                     cmd = new SqlCommand(cmdTextStr, con);
                 }
 
+                
                 SqlDataReader dr;
                 dr = cmd.ExecuteReader();
                 DataTable dt = new DataTable();
                 dt.Load(dr);
                 dgvBooks.DataSource = dt;
+
+                //Check if search key is not found and display error message.
+                if (dt != null)
+                {
+                    if (dt.Rows.Count == 0)
+                    {
+                        MessageBox.Show("Search key not found. Try again.", "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
+                    }
+                }
+                
                 con.Close();
-            }            
+            }
+            //reset searchbox text to placeholder
```

Comment:
Suggestion: remove code to reset txtSearch.Text to placeholder text so the user can still see what they entered just in case they want to check if they made a typo or an error